src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala

```diff
@@ -0,0 +1,194 @@
+  )
+
+  // TODO: create helper method to augment Settings object with data from Cassandra
+  // TODO: move most of these settings to Cassandra
+  private implicit object Settings extends Settings {
+    import scala.util.Properties.envOrNone
+
+    val progressDir: String = envOrFail(Constants.Env.HighlyAvailableProgressDir)
+    val featureServiceHost: String = envOrFail(Constants.Env.FeatureServiceHost)
+    val oxfordLanguageToken: String = envOrFail(Constants.Env.OxfordLanguageToken)
+    val oxfordVisionToken: String = envOrFail(Constants.Env.OxfordVisionToken)
+
+    val appInsightsKey: Option[String] = envOrNone(Constants.Env.AppInsightsKey)
+    val modelsDir: Option[String] = envOrNone(Constants.Env.LanguageModelDir)
+
+    private def envOrFail(name: String): String = {
+      envOrNone(name) match {
+        case Some(v) => v
+        case None =>
+          DriverLog.error(s"Environment variable not defined: $name")
+          sys.exit(1)
+      }
+    }
+  }
+
+  // TODO: logging configuration should be done in log4j config file
+  AppInsights.init(Settings.appInsightsKey)
+
+  Logger.getLogger("org").setLevel(Level.ERROR)
+  Logger.getLogger("akka").setLevel(Level.ERROR)
+  Logger.getLogger("libinstagram").setLevel(Level.DEBUG)
+  Logger.getLogger("libfacebook").setLevel(Level.DEBUG)
+  Logger.getLogger("liblocations").setLevel(Level.DEBUG)
+
+  private implicit object TransformContext extends TransformContext {
+    val geofence = Geofence(north = 49.6185146245, west = -124.9578052195, south = 46.8691952854, east = -121.0945042053)
+    val placeRecognizer = new PlaceRecognizer(Settings.modelsDir)
+    val featureServiceClient = new FeatureServiceClient(Settings.featureServiceHost)
+    val locationsExtractor = new LocationsExtractor(featureServiceClient, geofence, Some(placeRecognizer)).buildLookup()
+    val keywordExtractor = new KeywordExtractor(List("Ariana"))
+    val imageAnalyzer = new ImageAnalyzer(ImageAnalysisAuth(Settings.oxfordVisionToken), featureServiceClient)
+    val languageDetector = new LanguageDetector(LanguageDetectorAuth(Settings.oxfordLanguageToken))
+    val sentimentDetector = new SentimentDetector(SentimentDetectorAuth(Settings.oxfordLanguageToken))
+    val supportedLanguages = Set("en", "fr", "de")
+  }
+
+  private def createStreamingContext(): StreamingContext = {
+    val conf = new SparkConf()
+      .setAppName(Constants.SparkAppName)
+      .setIfMissing("spark.master", Constants.SparkMasterDefault)
+
+    implicit val streamingContext = new StreamingContext(
+      new SparkContext(conf),
+      Seconds(Constants.SparkStreamingBatchSizeDefault))
+
+    val streamProvider = StreamProviderFactory.create()
+    val streamRegistry = getStreamRegistry
+
+    // Attach each pipeline (aka code path)
+    // 'fortisEvents' is the stream of analyzed data aggregated (union) from all pipelines
+    val fortisEvents = pipelines.flatMap(
+      pipeline => pipeline(streamProvider, streamRegistry)
```

Comment:
Wouldn't `_(streamProvider, streamRegistry)` also work here? If we're already using lots of underscores, might as well be consistent.